packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -104,8 +105,28 @@ export type SdkType =
 export interface SdkBuiltInType extends SdkTypeBase {
   kind: SdkBuiltInKinds;
   encode: string;
+  name: string;
+  tspNamespace?: string;
+  baseType?: SdkBuiltInType;
 }
 
+type TypeEquality<T, U> = keyof T extends keyof U
+  ? keyof U extends keyof T
+    ? true
+    : false
+  : false;
+
+// these two vars are used to validate whether our SdkBuiltInKinds are exhaustive for all possible values from typespec
+// if it is not, a typescript compilation error will be thrown here.
+const _: TypeEquality<Exclude<SupportedBuiltInKinds, SdkBuiltInKinds>, never> = true;
```

Comment:
is there a clearer way of writing this? It's quite hard to read.

Reply:
I think this is not here for readability. I added this because I think it is necessary to validate our current kind defined in those enums are exhaustive for the kinds defined in std libraries.

We should remove these checks once we get rid of those enums.

I predicated that there should be a readability issue, therefore I added the above comments explaining what these two lines are doing. Please let me know if this still not clear enough

Reply:
A good use of `never` :-)